added %ARCHITECTURE% to package name

mpanighetti opened the pull request:
- added `%ARCHITECTURE%` to package name for differentiation
- removed redundant `%pkg_path%` definition from Eclipse.install.recipe

homebysix commented:
Does this change work for you? I'm getting an error:

```
The following recipes failed:
    Eclipse/Eclipse.pkg.recipe
        Error in com.github.homebysix.pkg.Eclipse: Processor: AppPkgCreator: Error: Can't stat %RECIPE_CACHE_DIR%: [Errno 2] No such file or directory: '%RECIPE_CACHE_DIR%'
```

mpanighetti commented:
Odd, I'm getting the same error, seems to be an issue with the custom `pkg_path` being set. `AppPkgCreator` works fine if I stick with the default file path. Guessing that's some kind of bug in the processor since I can see `RECIPE_CACHE_DIR` is getting read properly in the recipe run.

As a workaround, I could just rename the file with `FileMover` to achieve the desired result; folks would just need to explicitly define `pkg_path` in any subsequent recipes.

mpanighetti commented:
Reverted to `PkgCreator` and updated `%pkgname%` to include `%ARCHITECTURE%` which works in my testing and achieves the desired effect of avoiding name collision when running both Apple Silicon and Intel recipe overrides.
